src/components/FeedInfo.tsx

```diff
@@ -76,36 +77,55 @@ export class EditFeed extends React.Component<DispatchProps & StateProps, EditFe
                 checked: true,
                 loading: false,
             });
-            await this.onAdd(feed);
+            this.onAdd(feed);
+            await this.props.onDownloadFeed(feed);
```

Comment:
This is not necessary, because the onAddFeed implementation in the container also triggers the download.

If you want to keep it for some reason, `await` here is not necessary, because it calls the onDownloadFeed which is not async.

Reply:
Good catch!
about await, I did not look into it just followed what you did :-)
Which is await when the dispatched action is async, are you saying that does not work like that?

Reply:
Here I just stated that the `onDownloadFeed` function is by definition is not an async function in the `DispatchProps` so no `await` is needed.

It doesn't matter if the dispatched action is async or not from the perspective of the function that is calling the `dispatch`, because `dispatch` is not an async function. I don't know how it's implemented exactly, but I suspect that the async actions are put in a queue and executed later.